<launch>

    <arg name="markerId"        default="5"/>
    <arg name="markerSize"      default="0.20"/>    <!-- in m -->
    <arg name="eye"             default="right"/>
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->
    <!-- <arg name="marker_frame"    default="aruco_marker_frame"/> -->
    <!-- <arg name="marker_frame"    default="map"/> -->
    <!-- <arg name="ref_frame"       default=""/>  leave empty and the pose will be published wrt param parent_name -->

    <param name="use_sim_time" value="true"/>

    <node name="republish_camera_image"
          pkg="image_transport"
          type="republish"
          args="_image_transport:=compressed in:=/Player0/camera/image raw out:=/Player0/camera/uncompressed"
          output="screen">
    </node>

    <!-- Working -->
    <!--<node pkg="pose_transformer" type="test_new_transformer_node.py" name="transformer_node" output="screen"/> -->
    
    <!-- Testing -->
    <!-- <node pkg="pose_transformer" type="test_new_transformer_node_copy.py" name="transformer_node" output="screen"/> -->

    <!-- <node pkg="pose_transformer" type="tf_error_calc.py" name="tf_error_calc_node" output="screen"/> -->

    <node pkg="pose_transformer" type="test_drone_vio_transformer_node.py" name="transformer_node" output="screen"/>

    <!-- Leaving reference_frame empty to publish Hl2_camera -> aruco_marker_hl2_frame -->
    <node pkg="aruco_ros" type="single" name="aruco_hl2">
        <remap from="/camera_info" to="/Player0/camera/camera_info/updated" />
        <remap from="/image" to="/Player0/camera/uncompressed" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value=""/>
        <param name="camera_frame"       value="hl2_camera"/>
        <param name="marker_frame"       value="aruco_marker_hl2_frame" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

    <!-- Se refterence_frame = aruco_marker_frame to publish aruco_marker_frame -> base_link -->
    <node pkg="aruco_ros" type="single" name="aruco_drone">
        <remap from="/camera_info" to="/camera/camera_info" />
        <remap from="/image" to="/camera/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value=""/>
        <param name="camera_frame"       value="base_link"/>
        <param name="marker_frame"       value="aruco_marker_drone_frame" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

</launch>

